### ISSUE TITLE
Update MPS Doc to Explain Branch Matching with O3DE

### ISSUE THREAD

**Issue by AMZN-Gene** [o3de/o3de-multiplayersample#184]
**Is your feature request related to a problem? Please describe.**
Sometimes I match MPS branch with the wrong O3DE branch. For example, I'll use MPS Main branch, with the latest O3DE dev branch

**Describe the solution you'd like**
Add a note to the docs explaining branches need to match.

**Comment by willihay** [comment]
Yes! I made this n00b error too the first time I tried to build the sample. This should be noted in this repo's readme and maybe in the tutorial, too.
